cmake/modules/AddSwift.cmake

```diff
@@ -638,6 +638,7 @@ endmacro()
 
 # Declare that files in this library are built with LLVM's support
 # libraries available.
-macro(set_swift_llvm_is_available)
-  add_compile_options(-DSWIFT_LLVM_SUPPORT_IS_AVAILABLE)
+macro(set_swift_llvm_is_available name)
+  target_compile_options(${name} PRIVATE
+    $<$<COMPILE_LANGUAGE:C,CXX,OBJC,OBJCXX>:-DSWIFT_LLVM_SUPPORT_IS_AVAILABLE>)
```

Comment:
This really should be `target_compile_definitions` and just drop the `-D` prefix.  However, this is a very odd thing to have to set on a per target basis.  Why is this needed?

Reply:
I don't know. I am just refactoring.

Reply:
I'll change it to target_compile_definition.